x-pack/plugins/observability/server/services/slo/tasks/orphan_summary_cleanup_task.ts

```diff
@@ -0,0 +1,238 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ElasticsearchClient, Logger } from '@kbn/core/server';
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { AggregationsCompositeAggregateKey } from '@elastic/elasticsearch/lib/api/types';
+import { ALL_SPACES_ID } from '@kbn/spaces-plugin/common/constants';
+import { SLORepository } from '..';
+import { ObservabilityConfig } from '../../..';
+import { SLO_SUMMARY_DESTINATION_INDEX_PATTERN } from '../../../../common/slo/constants';
+
+export const TASK_TYPE = 'SLO:ORPHAN_SUMMARIES-CLEANUP-TASK';
+
+export const getDeleteQueryFilter = (
+  sloSummaryIdsToDelete: Array<{ id: string; revision?: number }>
+) => {
+  return sloSummaryIdsToDelete.map(({ id, revision }) => {
+    return {
+      bool: {
+        must: [
+          {
+            term: {
+              'slo.id': id,
+            },
+          },
+          {
+            term: {
+              'slo.revision': Number(revision),
+            },
+          },
+        ],
+      },
+    };
+  });
+};
+
+export class SloOrphanSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private taskManager?: TaskManagerStartContract;
+  private repository?: SLORepository;
+  private esClient?: ElasticsearchClient;
+  private config: ObservabilityConfig;
+
+  constructor(taskManager: TaskManagerSetupContract, logger: Logger, config: ObservabilityConfig) {
+    this.logger = logger;
+    this.config = config;
+
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'SLO Definitions Cleanup Task',
+        timeout: '3m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              return this.runTask();
+            },
+
+            cancel: async () => {
+              this.abortController.abort('[SLO] Definitions clean up Task timed out');
+            },
+          };
+        },
+      },
+    });
+  }
+
+  runTask = async () => {
+    const runAt = new Date().toISOString();
+
+    if (this.repository && this.esClient) {
+      let searchAfterKey: AggregationsCompositeAggregateKey | undefined;
+
+      do {
+        const { sloSummaryIds, searchAfter } = await this.fetchSloSummariesIds(searchAfterKey);
+
+        if (sloSummaryIds.length === 0) {
+          return { state: { lastRunAt: runAt } };
+        }
+
+        searchAfterKey = searchAfter;
+        const sloDefinitions = await this.repository.findAllByIds(
+          sloSummaryIds.map(({ id }) => id),
+          ALL_SPACES_ID
+        );
```

Comment:
Using the repository will load the full SLO model + doing some io-ts schema decoding for each vs doing an ad-hoc query that would load just the [id, revision]. 
Should be ok, but just want to make sure we understand the implication